src/Upgrade/Application/Strategy/AbstractStrategy.php

```diff
@@ -39,6 +46,9 @@ public function upgrade(): StepsResponseDto
 
             $stepsExecutionDto = $step->run($stepsExecutionDto);
 
+            if (!$stepsExecutionDto->getIsSuccessful()) {
```

Comment:
1. Why is it in the separate `if`. Exactly the same condition we have below this one.
2. How many steps require code fix? If it is only one, maybe we can use code fixer only in it?

Reply:
no Everything is ok here. if the step is failed we try to fix it. in the fixer loop, we look for the applicable one and try to fix the step. if we fix the step we break from the loop and skip rollback, if not we run rollback. everything is clear here